### ISSUE TITLE
[QUESTION]: adding a new spell group

### ISSUE THREAD

**Issue by Sillsoft42** [Larkinabout/fvtt-token-action-hud-core#272]
I'm using a class from a module (Apothecary from Sebastian Crowe's Guide to Drakkenheim) and it has a different spell slot system setup.  Is there a way I can get it so the spells I have placed on my character show up in the token action hud?

System: dnd5e 4.0.4

**Comment by Larkinabout** [comment]
Oh, it's a premium module, so they're probably not going to add support for Token Action HUD. You might need to create a separate module to do it, but it depends in what their class implementation is actually doing and how it's storing the spells in the actor. Unfortunately, I can't support it myself as I don't own the module.

**Comment by Sillsoft42** [comment]
Yeah, that's what I figured.  I have my own personal module that I could probably use to create the hooks.  Could you tell me which hooks I would need to use as a starting point.  

Looking at the actor object, they add the following item to the system.spells object on the actor.
{
    "type": "apothecary",
    "level": 1,
    "max": 1,
    "value": null
}

**Comment by Sillsoft42** [comment]
I tried messing around today using the system template example and I haven't been able to make it very far so I'm probably doing something wrong.  At one point, I did seem to get a new group to show up but now I can't even get that working again.

**Comment by Larkinabout** [comment]
I'll have to have a think how to do it on top of the system module. Ultimately, you'd need to add your group on the `tokenActionHudCoreRegisterDefaults`, add an extender of the action handler on `tokenActionHudCoreAddActionHandlerExtenders`, then also add an extender for the roll handler. The hook for the latter doesn't exist yet, but it's in development for the next release.

**Comment by Sillsoft42** [comment]
I appreciate you looking into it.  TAH is such a great module.

**Comment by Sillsoft42** [comment]
So I'm taking another run at doing this and I've figured out how to add a new group but I'm stuck on the next steps.

I also don't fully understand what the 'action handler' does and what the 'roll handler' does

Which bit is the part where I look at an actor and figure out what items show up in this new group that I added.

**Comment by madseumas** [comment]
If it still helps the action handler is where you look at an actor and figure out how to make items show up and the roll handler is where you take the encoded value from the event generated by someone clicking on a button and have it do the roll.
